src/controller/CHIPDeviceController.cpp

```diff
@@ -2093,7 +2091,37 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     case CommissioningStage::kSecurePairing:
         break;
     }
-} // namespace Controller
+}
+
+CHIP_ERROR DeviceController::UpdateDevice(NodeId deviceId)
+{
+    VerifyOrReturnError(mState == State::Initialized && mFabricInfo != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    OperationalDeviceProxy * proxy = GetDeviceSession(mFabricInfo->GetPeerIdForNode(deviceId));
```

Comment:
No, this part is still broken compared to what we used to have.

What we used to have is that:
1.  We would re-resolve the address.
2. We would try to re-establish CASE (if `this` is a DeviceCommissioner).

The new code first tries to re-establish CASE (to the old address, if we have one) and then kicks off a lookup of the new address.  Which is not really going to work: the CASE will be trying to be established to the wrong place.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/16142 to track fixing this.